src/components/molecules/AnnouncementMessage/AnnouncementMessage.tsx

```diff
@@ -1,53 +1,106 @@
-import React, { useCallback, useEffect, useState } from "react";
-import classNames from "classnames";
+import React, { useEffect } from "react";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faTimesCircle } from "@fortawesome/free-solid-svg-icons";
+import classNames from "classnames";
+
+import { Banner } from "types/banner";
 
-import { useChatSidebarControls } from "hooks/chatSidebar";
+import { isDefined } from "utils/types";
+
+import { useShowHide } from "hooks/useShowHide";
 
 import { RenderMarkdown } from "components/organisms/RenderMarkdown";
+import { LinkButton } from "components/atoms/LinkButton";
 
 import "./AnnouncementMessage.scss";
 
-type AnnouncementMessageProps = {
-  message?: string;
-};
+export interface AnnouncementMessageProps {
+  banner: Banner;
```

Comment:
This field is not optional but this file is using it in a lot of places with the optional syntax `?`. You can remove that optional operator across the usages.

Reply:
I'm not seeing any place where `banner` is used with optional syntax

Reply:
added `| undefined` type as banner can be undefined 947713e